File: FrontEnd/domain/entities/Incident.ts
export type IncidentCategory = 'Maintenance' | 'Housekeeping' | 'Guest Complaint' | 'Security' | 'IT' | 'Other';
export type IncidentPriority = 'Low' | 'Medium' | 'High' | 'Critical';
export type IncidentStatus = 'Open' | 'In Progress' | 'Resolved' | 'Closed';

export interface Incident {
  id: string;
  category: IncidentCategory;
  subject: string;
  room: string;
  bookingRef?: string;
  guestName?: string;
  priority: IncidentPriority;
  status: IncidentStatus;
  reportedBy: string;
  assignedTo: string;
  createdAt: string;
  updatedAt: string;
  slaBreached: boolean;
  description: string;
}

File: BackEnd/app/models/incident.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database import Base


class Incident(Base):
    __tablename__ = "incidents"

    id = Column(Integer, primary_key=True, index=True)
    subject = Column(String, index=True)
    description = Column(String)
    room = Column(String)  # Room Number/ID
    priority = Column(String)  # Low, Medium, High, Critical
    status = Column(String, default="Open")  # Open, In Progress, Resolved, Closed
    category = Column(String)  # Maintenance, Housekeeping, etc.
    guest_name = Column(String, nullable=True)
    reported_by = Column(String, nullable=True)
    assigned_to = Column(String, nullable=True)
    created_at = Column(String)  # ISO Date String
    updated_at = Column(String)  # ISO Date String
    sla_breached = Column(Boolean, default=False)

    tenant_id = Column(UUID(as_uuid=True), ForeignKey("tenants.id"))

    tenant = relationship("Tenant")
